### PR TITLE
feat: external signer to read encrypted privkeys

### PR DESCRIPTION
**Pull request by DariusParvin**
- When running in external signer mode, express will expect the private
key json file to be in the encrypted format (same as on the keycard but
with escape characters).
- Express expects the wallet passphrase to be set in the environment
  with the name `WALLET_<walletId>_PASSPHRASE`
- Update documentation and move to new file EXTERNAL_SIGNER.md

Ticket: BG-43925

### CODE REVIEW

File: modules/express/src/clientRoutes.ts [hunk 1]
```diff
@@ -366,19 +366,46 @@ function handleCanonicalAddress(req: express.Request) {
...
+  try {
+    const decrypted = bg.decrypt({ password: walletPw, input: encryptedPrivKey });
+    return decrypted;
+  } catch (e) {
+    throw new Error(`Error when trying to decrypt private key: ${e}`);
   }
+}
+
+export async function handleV2Sign(req: express.Request) {
+  express;
```
Comment (tylerlevine) [inline modules/express/src/clientRoutes.ts:398]: what's this?

File: modules/express/src/clientRoutes.ts [hunk 2]
```diff
@@ -366,19 +366,46 @@ function handleCanonicalAddress(req: express.Request) {
...
-    throw new Error(`Could not find a field for walletId: ${walletId} in ${req.config.signerFileSystemPath}`);
+  const encryptedPrivKey = JSON.parse(privKeyFile);
+  if (encryptedPrivKey[walletId] === undefined) {
+    throw new Error(`Could not find a field for walletId: ${walletId} in ${path}`);
+  }
+  return encryptedPrivKey[walletId];
+}
+
+function decryptPrivKey(encryptedPrivKey: string, walletPw: string): string {
+  const bg = new BitGo();
```
Comment (tylerlevine) [inline modules/express/src/clientRoutes.ts:388]: pass bitgo object in as param - it's available in the calling scope as `req.bitgo` (see line 405)

File: modules/express/src/clientRoutes.ts [hunk 3]
```diff
@@ -366,19 +366,46 @@ function handleCanonicalAddress(req: express.Request) {
...
+  try {
+    const decrypted = bg.decrypt({ password: walletPw, input: encryptedPrivKey });
+    return decrypted;
```
Comment (tylerlevine) [inline modules/express/src/clientRoutes.ts:391]:
```suggestion
    return bg.decrypt({ password: walletPw, input: encryptedPrivKey });
```

File: modules/express/src/clientRoutes.ts [hunk 4]
```diff
@@ -366,19 +366,46 @@ function handleCanonicalAddress(req: express.Request) {
...
+  const walletPw = getWalletPwFromEnv(walletId);
+  const privKeyPath = req.config.signerFileSystemPath;
+  assert(typeof privKeyPath === 'string');
+  const encryptedPrivKey = await getEncryptedPrivKey(privKeyPath, walletId);
+  const privKey = decryptPrivKey(encryptedPrivKey, walletPw);
   const bitgo = req.bitgo;
   const coin = bitgo.coin(req.params.coin);
   try {
-    return await coin.signTransaction({ ...req.body, ...{ prv: privKey[walletId] } });
+    return await coin.signTransaction({ ...req.body, ...{ prv: privKey } });
```
Comment (tylerlevine) [inline modules/express/src/clientRoutes.ts:408]: literal destruction is redundant here
```suggestion
    return await coin.signTransaction({ ...req.body, prv: privKey });
```

**Review by tylerlevine — CHANGES_REQUESTED** [review]
good work, just a few comments

File: modules/express/src/clientRoutes.ts (hunk 1 above)
Comment (DariusParvin) [reply modules/express/src/clientRoutes.ts]: that was unintentional. removed

**Comment by DariusParvin** [comment]
Good catches, thanks for the feedback. I've addressed them now

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**